spec/views/hq/sponsors/_form_spec.rb

```diff
@@ -24,129 +23,95 @@ def render_sponsor_form current_sponsor
-      end
+      expect(rendered).to have_select("Status", selected: sponsor_full.status.name, options: Status.pluck(:name))
 
-      assert_select "select#sponsor_gender" do
-        assert_select "option", value: Settings.lookup.gender.first,
-                                html: CGI::escape_html(Settings.lookup.gender.first)
-      end
+      expect(rendered).to have_select("Gender", selected: sponsor_full.gender, options: Settings.lookup.gender)
 
-      assert_select "input#sponsor_request_fulfilled" do
-        assert_select "[disabled=?]", "disabled"
-        assert_select "[checked]", sponsor_full.request_fulfilled
+      if sponsor_full.request_fulfilled
+        expect(rendered).to have_checked_field("Request fulfilled", disabled: true)
+      else
+        expect(rendered).to have_unchecked_field("Request fulfilled", disabled: true)
       end
 
-      assert_select "select#sponsor_sponsor_type_id" do
-        assert_select "[disabled=?]", "disabled"
-        assert_select "option", value: sponsor_full.sponsor_type_id,
-          html: CGI::escape_html(sponsor_full.sponsor_type.name) do
-            assert_select "[selected=?]", "selected"
-        end
-      end
+      expect(rendered).to have_select("sponsor_sponsor_type_id", selected: sponsor_full.sponsor_type.name,
+                                       options: SponsorType.pluck(:name), disabled: true)
 
-      assert_select "select#sponsor_organization_id" do
-        assert_select "[disabled=?]", "disabled"
-        assert_select "option", value: sponsor_full.organization_id
-        if sponsor_full.organization
-          assert_select "option",
-            html: CGI::escape_html(sponsor_full.organization.name)
-        else
-          assert_select "option", value: false
-        end
+      if sponsor_full.organization
+        expect(rendered).to have_selector("select[id='sponsor_organization_id'][disabled] option[selected]", text: sponsor_full.organization.name)
+      else
+        expect(rendered).to have_selector("select[id='sponsor_organization_id'][disabled] option", text: "")
+        expect(rendered).to_not have_selector("select[id='sponsor_organization_id'][disabled] option[selected]")
       end
 
-      assert_select "select#sponsor_branch_id" do
-        assert_select "[disabled=?]", "disabled"
-        assert_select "option", value: sponsor_full.branch_id
-        if sponsor_full.branch
-          assert_select "option",
-            html: CGI::escape_html(sponsor_full.branch.name)
-        else
-          assert_select "option", value: false
-        end
+      if sponsor_full.branch
+        expect(rendered).to have_selector("select[id='sponsor_branch_id'][disabled] option[selected]", text: sponsor_full.branch.name)
+      else
+        expect(rendered).to have_selector("select[id='sponsor_branch_id'][disabled] option", text: "")
+        expect(rendered).to_not have_selector("select[id='sponsor_branch_id'][disabled] option[selected]")
```

Comment:
It seems that this ends up testing two separate things - the fact that fields are correctly pre-filled with object's attribute values (the form's responsibility) and that organization and branch cannot co-exist for the same sponsor (the model's responsibility). Shouldn't we be able to rely on the model to do its job and only test that if a branch is specified, the relevant field is pre-filled, and likewise for organization?

Also, since both the organization and the branch selectors should be disabled regardless of which affiliation a sponsor has, I think this ought to be tested outside of any `if`.